sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -0,0 +1,410 @@
+
+    def _safe_get_metric_and_columns(
+            self, column_name_1, column_name_2, real_data, synthetic_data, metadata):
+        """Call ``_get_metric_and_columns`` and catch any exceptions.
+
+        Args:
+            column_name_1 (str):
+                The name of the first column
+            column_name_2 (str):
+                The name of the second column
+            real_data (pandas.DataFrame):
+                The real data
+            synthetic_data (pandas.DataFrame):
+                The synthetic data
+            metadata (dict):
+                The metadata of the table
+        """
+        try:
+            return self._get_metric_and_columns(
+                column_name_1, column_name_2, real_data, synthetic_data, metadata
+            )
+        except Exception:
+            return None, None, None
+
+    def _safe_compute_breakdown(
+            self, column_name_1, column_name_2, metric, columns_real, columns_synthetic):
+        """Call ``metric.compute_breakdown`` and catch any exceptions.
+
+        Args:
+            column_name_1 (str):
+                The name of the first column
+            column_name_2 (str):
+                The name of the second column
+            metric (Metric):
+                The metric to use
+            columns_real (pandas.DataFrame):
+                The real data
+            columns_synthetic (pandas.DataFrame):
+                The synthetic data
+        """
+        try:
+            return metric.compute_breakdown(
+                real_data=columns_real, synthetic_data=columns_synthetic
+            )
+        except Exception as e:
+            message = (
+                f"Unable to compute Column Pair Trends for column ('{column_name_1}', "
+                f"'{column_name_2}'). Encountered Error: {type(e).__name__} {e}"
+            )
+            self._warning_messages_list.append(message)
+            return {'score': np.nan, 'real': np.nan, 'synthetic': np.nan}
+
+    def _handling_warning_messages(self):
+        """Handle warning messages.
+
+        Remove duplicate messages and raise the warnings.
+        """
+        if self._warning_messages_list:
+            self._warning_messages_list = list(dict.fromkeys(self._warning_messages_list).keys())
+            for message in self._warning_messages_list:
+                warnings.warn(message)
+
+    def _generate_details(self, real_data, synthetic_data, metadata, progress_bar):
+        """Generate the _details dataframe for the column pair trends property.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data
+            synthetic_data (pandas.DataFrame):
+                The synthetic data
+            metadata (dict):
+                The metadata of the table
+            progress_bar:
+                The progress bar to use. Defaults to tqdm.
+        """
+        processed_real_data, processed_synthetic_data = self._get_processed_data(
```

Comment:
I wonder if it would be simpler to do the following steps:
1. Process the data (get rid of datetimes)
2. Get the discretized version of the processed data as a separate dataframe (that way you don't need to make unique names)
3. In the loop below, figure out if either column is discrete
4. Based on that, either pass the regular or discretized data to the metric